mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/dto/DeletedMessageWithStorageInformationConverter.java

```diff
@@ -0,0 +1,94 @@
+import org.apache.james.vault.metadata.StorageInformation;
+
+import com.github.fge.lambdas.Throwing;
+import com.github.steveash.guavate.Guavate;
+import com.google.common.collect.ImmutableList;
+
+public class DeletedMessageWithStorageInformationConverter {
+    private final BlobId.Factory blobFactory;
+    private final MessageId.Factory messageIdFactory;
+    private final MailboxId.Factory mailboxIdFactory;
+
+    @Inject
+    public DeletedMessageWithStorageInformationConverter(BlobId.Factory blobFactory,
+                                                         MessageId.Factory messageIdFactory,
+                                                         MailboxId.Factory mailboxIdFactory) {
+        this.blobFactory = blobFactory;
+        this.messageIdFactory = messageIdFactory;
+        this.mailboxIdFactory = mailboxIdFactory;
+    }
+
+    public StorageInformation toDomainObject(DeletedMessageWithStorageInformationDTO.StorageInformationDTO storageInformationDTO) {
+        return StorageInformation.builder()
+            .bucketName(BucketName.of(storageInformationDTO.getBucketName()))
+            .blobId(blobFactory.from(storageInformationDTO.getBlobId()));
+    }
+
+    public DeletedMessage toDomainObject(DeletedMessageWithStorageInformationDTO.DeletedMessageDTO deletedMessageDTO) {
+        return DeletedMessage.builder()
+            .messageId(messageIdFactory.fromString(deletedMessageDTO.getMessageId()))
+            .originMailboxes(deserializeOriginMailboxes(deletedMessageDTO.getOriginMailboxes()))
+            .user(User.fromUsername(deletedMessageDTO.getOwner()))
+            .deliveryDate(ZonedDateTime.parse(deletedMessageDTO.getDeliveryDate()))
+            .deletionDate(ZonedDateTime.parse(deletedMessageDTO.getDeletionDate()))
+            .sender(MaybeSender.getMailSender(deletedMessageDTO.getSender()))
+            .recipients(deserializeRecipients(deletedMessageDTO.getRecipients()))
+            .hasAttachment(deletedMessageDTO.getHasAttachment())
+            .size(deletedMessageDTO.getSize())
+            .subject(deletedMessageDTO.getSubject())
+            .build();
+    }
+
+    public DeletedMessageWithStorageInformation toDomainObject(DeletedMessageWithStorageInformationDTO deletedMessageWithStorageInfoDTO) {
+        return new DeletedMessageWithStorageInformation(
+            toDomainObject(deletedMessageWithStorageInfoDTO.getDeletedMessage()),
+            toDomainObject(deletedMessageWithStorageInfoDTO.getStorageInformation()));
+    }
+
+    private ImmutableList<MailboxId> deserializeOriginMailboxes(List<String> originMailboxes) {
+        return originMailboxes.stream()
+            .map(mailboxId -> mailboxIdFactory.fromString(mailboxId))
+            .collect(Guavate.toImmutableList());
+    }
+
+    private ImmutableList<MailAddress> deserializeRecipients(List<String> recipients) {
+        return recipients.stream()
+            .map(Throwing.function(recipient -> new MailAddress(recipient)))
```

Comment:
sneakyThrow?

Reply:
What would it bring?

Reply:
we do it everywhere else, why making an exception here?

Reply:
Well because explicit exception handling is verbose while throwing runtime is not.

What's important is: do we get notified when something goes wrong? (yes in both case)

Do we abort a complex computation for a single error? (That's the responsibility from my Cassandra code after all, and explicit exception handling would force me to handle it)